src/Engines/WebGPU/Extensions/engine.externalTexture.ts

```diff
@@ -0,0 +1,19 @@
+import { WebGPUEngine } from "../../webgpuEngine";
+import { ExternalTexture } from "../../../Materials/Textures/externalTexture";
+import { Nullable } from "../../../types";
+import { WebGPUExternalTexture } from "../webgpuExternalTexture";
+
+WebGPUEngine.prototype.createExternalTexture = function (video: HTMLVideoElement): Nullable<ExternalTexture> {
+    const texture = new WebGPUExternalTexture(video);
+    return texture;
+};
+
+WebGPUEngine.prototype.setExternalTexture = function (name: string, texture: Nullable<ExternalTexture>): void {
+    if (this._currentEffect) {
+        if (!texture) {
+            this._currentMaterialContext.setTexture(name, null);
+            return;
+        }
+        this._setInternalTexture(name, texture);

```

Comment:
Yes, it can be called multiple times. But in any case it's up to the user to handle the lifetime of their external textures (knowing that in WebGPU you can't explicitly delete external textures anyway).